src/system/mod.rs

```diff
@@ -655,7 +655,7 @@ fn read_proc_stat<T: FromStr>(pid: WithProcess, field_idx: isize) -> io::Result<
     // we've now passed the first two fields, so we are at index 1, now we skip over
     // fields until we arrive at the field we are searching for
     let mut curr_field = 1;
-    while curr_field < field_idx && !stat.is_empty() {
```

Comment:
We changed the 0- or 1-based nature of this function based on that change, but this function does not have a documenting comment; we should add a comment at the start of this `fn` to describe what it does. 
(I.e. "fetches the N'th field from /proc/stat on Linux" or something)

This is 'ad hoc parsing code' and that's always a bit iffy to understand some time after it was written.

Reply:
Added a doc comment which also links to the kernel docs listing all available fields.

Reply:
I'm looking at this change again: `field_idx` is one *higher* than in the pre-#887 situation, but at the same time the new loop also iterates one iteration *more*. So aren't we reading completely different fields (i.e. two fields further to the right).

Perhaps my tired eyes deceive me, or perhaps we need some better unit tests. :-)

Reply:
My tired eyes did not deceive. Nicely caught by our compliance testing framework, but I'll be introducing a unit test for `read_proc_stat` in a followup. :-)